react/getting_started_with_react/keys_in_react.md

```diff
@@ -0,0 +1,124 @@
+### Introduction
+
+In this lesson we will cover keys in React. Keys are special props for our components and we'll learn why they are used.
+
+### Lesson Overview
+
+By the end of the lesson you should be able to answer the following:
+
+- Why are keys and why does React need them. 
+- What are good and bad examples of keys. 
+
+### Why Does React Need Keys?
+
+In the upcoming lessons, as you learn more about the internal workings of React, more specifically, the rerendering process, you will understand the importance of keys. For now, we will keep it simple. Recall the previous lesson on rendering lists. We used the `.map()` method to iterate over an array of data and return a list of elements. Now imagine, if any of the item in the list were to change, how would React know which item to update?
+
+If the list where to change, one of two things _should_ happen:
+1. either, we completely re-render the entire list,
+2. or, hunt down that specific items that were changed and only re-render those.
+
+Assuming we want to hunt down that one specific item that was changed and NOT re-render the entire list. We need something to track that specific item. We can track down a specific item by using a `key`.
+
+When the list is updated for whatever reason, (either from a server or a user interaction), React matches the `key`s of each of the previous list to the updated list. If there were any changes, React will only update the items that have changed.
+
+As long as `keys` remain consistent and unique, React can handle the DOM effectively and efficiently.
+
+### Using Keys
+
+Keys are passed into the component or a DOM element as a prop. You should already be familiar with the syntax.
```

Comment:
should prop be marked in code as `prop` ? I haven't reviewed other lessons before so not sure if you decided on a standard here. With my limited knowledge on React this seems to be a react specific term. Similar to `argument` or `parameter`

Reply:
It doesn't really matter here, we could go with either honestly🤷